pandas/io/json.py

```diff
@@ -884,3 +940,175 @@ def _recursive_extract(data, path, seen_meta, level=0):
         result[k] = np.array(v).repeat(lengths)
 
     return result
+
+
```

Comment:
renaming atm is tricky as json_normalize is imported from here

so you could put the new code in say
pandas/io/jsons/ would work i think